### PR TITLE
Update install docs for NET6

### LINKED ISSUE #25358: MacOS and Linux uninstall instructions need updates for .NET 6
The `dotnet-core-uninstall` tool, the easiest mechanism for removing SDKs or runtimes on MacOS and Linux, does not support .NET 6. The instructions that are there for MacOS...

```sh
version="1.0.1"
sudo rm -rf /usr/local/share/dotnet/sdk/$version
sudo rm -rf /usr/local/share/dotnet/shared/Microsoft.NETCore.App/$version
sudo rm -rf /usr/local/share/dotnet/shared/Microsoft.AspNetCore.All/$version
sudo rm -rf /usr/local/share/dotnet/shared/Microsoft.AspNetCore.App/$version
sudo rm -rf /usr/local/share/dotnet/host/fxr/$version
```

...don't actually work for later SDKs and runtimes because there isn't one folder per version in these locations anymore. You have to specifically delete a targeted version in these locations which may not all be the same value. Doing what the instructions say here for, say, .NET 6, is going to cause errors and be confusing.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4efe9b33-ff9f-3b37-e8bd-a34c0a77bd1e
* Version Independent ID: 090b8fbd-39e6-b754-e1da-890b1123f2c4
* Content: [Remove the .NET runtime and SDK](https://docs.microsoft.com/en-us/dotnet/core/install/remove-runtime-sdk-versions)
* Content Source: [docs/core/install/remove-runtime-sdk-versions.md](https://github.com/dotnet/docs/blob/main/docs/core/install/remove-runtime-sdk-versions.md)
* Product: **dotnet**
* Technology: **dotnet-install**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### LINKED ISSUE #26531: Incorrect package dependencies?
The section https://docs.microsoft.com/en-us/dotnet/core/install/linux-debian#dependencies states that libgcc1 is required when you publish a self-contained app on Debian, but Debian 11 (at least) doesn't have this package?

The other packages exists, it's just this. I know about the libgcc_s1 package. Is this the same or is that something else?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c3310a1-6a2e-ca98-9618-8f3d729eafdd
* Version Independent ID: ab01f0aa-d1c0-df64-b0f9-ee211d45001f
* Content: [Install .NET on Debian - .NET](https://docs.microsoft.com/en-us/dotnet/core/install/linux-debian#debian-11-)
* Content Source: [docs/core/install/linux-debian.md](https://github.com/dotnet/docs/blob/main/docs/core/install/linux-debian.md)
* Product: **dotnet**
* Technology: **dotnet-install**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### LINKED ISSUE #26600: This page need update for 21.10
Ubuntu 21.10 is release last week. Able to use this docs without any trouble. It's just need to 21.10 on the page.
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 80707602-f283-e796-d3b8-3287fbc011dc
* Version Independent ID: 1d732b15-9e64-633a-000c-ad96d9d3818a
* Content: [Install .NET on Ubuntu - .NET](https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu)
* Content Source: [docs/core/install/linux-ubuntu.md](https://github.com/dotnet/docs/blob/main/docs/core/install/linux-ubuntu.md)
* Product: **dotnet**
* Technology: **dotnet-install**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Add 6.0 variant includes for linux install
- Update tables of supported OS
- Update each linux page
- Updated mac + related pages
- Updated windows
- Updated wording on upgrade assist docs

Fixes #25358
Fixes #26600
Fixes #26531
Contributes to #6446

### CODE REVIEW

**Comment by adegeo** [comment]
@tdykstra

File: docs/core/install/remove-runtime-sdk-versions.md [hunk 1]
```diff
@@ -77,6 +77,9 @@ sudo rm -rf /usr/share/dotnet/shared/Microsoft.AspNetCore.App/$version
 sudo rm -rf /usr/share/dotnet/host/fxr/$version
 ```
 
+> [!IMPORTANT]
+> The version folders may not match the "version" you're uninstalling. The individual runtimes and SDKs that are installed with the single .NET release may have different versions. For example, you may have installed ASP.NET Core 5 Runtime, which installed the 5.0.2 ASP.NET Core runtime and the 5.0.8 .NET runtime. Each have different versioned folders. For more information, see [Overview of how .NET is versioned](../versions/index.md).
```
Comment (tdykstra) [inline docs/core/install/remove-runtime-sdk-versions.md:81]:
```suggestion
> The version folders may not match the "version" you're uninstalling. The individual runtimes and SDKs that are installed with a single .NET release may have different versions. For example, you may have installed ASP.NET Core 5 Runtime, which installed the 5.0.2 ASP.NET Core runtime and the 5.0.8 .NET runtime. Each has a different versioned folder. For more information, see [Overview of how .NET is versioned](../versions/index.md).
```

**Review by tdykstra — APPROVED** [review]
Looks good, I left suggestions just for a few nits .

**Comment by adegeo** [comment]
@tdykstra Thanks for spotting those copy\paste errors. I'll update the 5.0 include files too.

**Pull request merged**